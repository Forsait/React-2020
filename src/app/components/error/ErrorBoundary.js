// @flow
import React from 'react';
import type { Node } from 'react';

type Props = {
  children: Node,
}

type State = {
  hasError: boolean,
}

export default class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;
  }
}
