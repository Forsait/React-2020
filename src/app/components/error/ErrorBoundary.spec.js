import React from 'react';
import { shallow } from 'enzyme';

import ErrorBoundary from './ErrorBoundary';

const NullComp = () => null;

describe('ErrorBoundary', () => {
  it('should display an ErrorMessage if wrapped component throws', () => {
    const com = shallow(
      <ErrorBoundary>
        <NullComp />
      </ErrorBoundary>,
    );

    const error = new Error('test');

    com.find(NullComp).simulateError(error);

    expect(com).toMatchSnapshot();
  });
});
