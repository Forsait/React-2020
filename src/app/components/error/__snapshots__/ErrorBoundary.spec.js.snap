// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should display an ErrorMessage if wrapped component throws 1`] = `
<h1>
  Something went wrong.
</h1>
`;
